src/libYARP_pcl/include/yarp/pcl/Pcl.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * Copyright (C) 2006-2010 RobotCub Consortium
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_PCL_PCL_H
+#define YARP_PCL_PCL_H
+
+#include <pcl/io/pcd_io.h>
+#include <yarp/sig/PointCloud.h>
+
+namespace yarp
+{
+namespace pcl
+{
+
+/**
+ * Convert a yarp::sig::PointCloud to a pcl::PointCloud object
+ * @param yarpCloud yarp::sig::PointCloud input
+ * @return a pcl PointCloud filled with data contained in the yarp cloud.
+ */
+template< class T1, class T2 >
+inline bool toPCL(yarp::sig::PointCloud< T1 > &yarpCloud, ::pcl::PointCloud< T2 > &pclCloud)
+{
+    yAssert(sizeof(T1) == sizeof(T2));
```

Comment:
[`static_assert`](http://en.cppreference.com/w/cpp/language/static_assert) ? 
As you are directly copyng the raw memory, perhaps it is even safer to just defined the template to operate on one consistent type.

Reply:
The problem here is that we can't have any control on `T2` because it is a pcl's type, the only thing that I can check is that the `sizeof` of the yarp type and the pcl type are the same.

I used `static_assert` inside the yarp PointCloud for defining which T1 are allowed and supported for now, the problem is that with the `static_assert` we can add the corresponding list of pcl supported types but inside these conversion functions (`toPcl` and `fromPcl`) we cannot check the correspondence between the yarp type and pcl type (e.g. `yarp::sig::XYZ_DATA` and `pcl::PointXYZ`

Note that it is possible that `sizeof(T1)==sizeof(T2)` but the two types can be incompatible.
Example:
`sizeof(yarp::sig::XYZ_DATA)` is equal to `sizeof(pcl::Normal)` but they have different fields.

This check is not sufficient to be 100% safe but better than nothing :sweat_smile: 
I hope that the user will these functions with the correct types and not try some exotic conversions like yarp XYZ to pcl NxNyNz (Normals):sweat_smile:

Reply:
I see, T1 is something like `struct vec3 { double x; double y; double z; }` and in memcpy you are relying of the memory structure of the two types to be the same, I get it. 
However, I think it is is desirable to transform this in `static_assert(sizeof(T1) == sizeof(T2))`.

Reply:
Ah ok now I get your point, yes probably is better `static_assert` in this case :+1: